apache/trafficcontrol
# Fix deliveryservice_stats API to use already-vendored influxdb client, fix RPM builds to discover all vendor dirs

## What does this PR (Pull Request) do?
Move the vendored influxdb library to the top-level vendor dir for TO and TS to share. Fix the deliveryservice_stats API to use the already-vendored library. Fix the rpm builds for TO, TM, and TS to properly discover all vendor dirs the repo and make the RPM builds fail if unvendored dependencies are added (except for the `golang.org/x/*` packages).



## Which Traffic Control components are affected by this PR?


- Traffic Ops
- Traffic Monitor
- Traffic Stats
- build

## What is the best way to verify this PR?
1. Build RPMs and verify CIAB starts up properly
```
cd infrastructure/cdn-in-a-box
make
docker-compose up --build
```
2. `GET /api/1.1/deliveryservice_stats`, make sure the endpoint still works
3. Run the traffic_ops_golang unit tests
4. For good measure, run the TO API tests and make sure they pass
5. verify RPM installation/upgrade/downgrade: install earlier versions of the rpms on a centos7 host, upgrade using the rpms built with this PR, downgrade, and upgrade

## The following criteria are ALL met by this PR

- [x] This PR fixes a build issue, no tests necessary
- [x] This PR fixes a build issue, no docs necessary
